### PR TITLE
Add possibility to control the truncation in bert score

### LINKED ISSUE #2673: truncation cause error while using `bert_score`
While using `bert_score`, I'm trying to use a model that restricted to 512 tokens, i.e. `BiomedNLP-PubMedBERT-base-uncased-abstract-fulltext`. But there's an error implying I'm trying to use larger tensor than 512 tokens, even though I set the `max_length` as something lower than `512`, that is `500`.

```
  File "/root/miniconda3/envs/pl/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/root/miniconda3/envs/pl/lib/python3.9/site-packages/torchmetrics/metric.py", line 236, in forward
    self._forward_cache = self._forward_reduce_state_update(*args, **kwargs)
  File "/root/miniconda3/envs/pl/lib/python3.9/site-packages/torchmetrics/metric.py", line 303, in _forward_reduce_state_update
    batch_val = self.compute()
  File "/root/miniconda3/envs/pl/lib/python3.9/site-packages/torchmetrics/metric.py", line 532, in wrapped_func
    value = compute(*args, **kwargs)
  File "/root/miniconda3/envs/pl/lib/python3.9/site-packages/torchmetrics/text/bert.py", line 221, in compute
    return bert_score(
  File "/root/miniconda3/envs/pl/lib/python3.9/site-packages/torchmetrics/functional/text/bert.py", line 414, in bert_score
    preds_embeddings, preds_idf_scale = _get_embeddings_and_idf_scale(
  File "/root/miniconda3/envs/pl/lib/python3.9/site-packages/torchmetrics/functional/text/bert.py", line 98, in _get_embeddings_and_idf_scale
    out = model(batch["input_ids"], batch["attention_mask"], output_hidden_states=True)
  File "/root/miniconda3/envs/pl/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/root/miniconda3/envs/pl/lib/python3.9/site-packages/transformers/models/bert/modeling_bert.py", line 979, in forward
    buffered_token_type_ids_expanded = buffered_token_type_ids.expand(batch_size, seq_length)
RuntimeError: The expanded size of the tensor (523) must match the existing size (512) at non-singleton dimension 1.  Target sizes: [1, 523].  Tensor sizes: [1, 512]
```

The snippets here:

```python
pred = ["abc "*2000]
gt = ["def "*2000]
bert_score = BERTScore(model_name_or_path="microsoft/BiomedNLP-BiomedBERT-base-uncased-abstract-fulltext", device="cuda:0", max_length=500, lang="en")
result = bert_score(pred, gt)
```

After detailed checking codes, the culprit is the following line, which setting `truncation` as `False`.

https://github.com/Lightning-AI/torchmetrics/blob/6a2fc72e86582c040573ccbcbd534819f23d162c/src/torchmetrics/text/bert.py#L213

May I ask why using this? Or could there be a flag that control whether truncate or not?

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes #2673

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
📚 Documentation preview 📚: https://torchmetrics--2776.org.readthedocs.build/en/2776/

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
@SkafteNicki looks good, lets just check the typing :)

File: src/torchmetrics/functional/text/helper_embedding_metric.py [hunk 1]
```diff
@@ -195,10 +195,11 @@ def __init__(
         tokenizer: Any,
         max_length: int = 512,
         preprocess_text_fn: Callable[
-            [List[str], Any, int], Union[Dict[str, Tensor], Tuple[Dict[str, Tensor], Optional[Tensor]]]
+            [List[str], Any, int.bool], Union[Dict[str, Tensor], Tuple[Dict[str, Tensor], Optional[Tensor]]]
```
Comment (Borda) [inline src/torchmetrics/functional/text/helper_embedding_metric.py:198]:
```suggestion
            [List[str], Any, int, bool], Union[Dict[str, Tensor], Tuple[Dict[str, Tensor], Optional[Tensor]]]
```

**Comment by Borda** [comment]
@SkafteNicki, do you have any idea why it takes so long on PT 1.13?
image

**Comment by SkafteNicki** [comment]
> @SkafteNicki, do you have any idea why it takes so long on PT 1.13? image

Not sure. If we just look at one of the new tests in this PR on 1.X it takes 75sec and in 2.X it takes 8sec. On my local laptop it takes around 4sec so it must be the 1.X build that something is wrong with.
What I can see is that all the tests that are really slow for the 1.X build is decorated with the `skip_on_connection_issues` decorator so I assume that maybe we are just seeing some connection issues for that build.

**Comment by SkafteNicki** [comment]
@Borda this time all of them took approximately the same amount of time so lets assume that it was a connection issue

**Pull request merged**